lms/djangoapps/experiments/utils.py

```diff
@@ -43,6 +45,53 @@ def check_and_get_upgrade_link_and_date(user, enrollment=None, course=None):
+    return json.loads(mock_programs_str)
+# TODO: clean up as part of REVEM-198 (END)
+
+
+# TODO: clean up as part of REVEM-199 (START)
+def is_enrolled_in_all_courses_in_program(courses_in_program, user_enrollments):
+    """
+    Determine if the user is enrolled in all courses in this program
+    """
+    # Get the enrollment course ids here, so we don't need to loop through them for every course run
+    enrollment_course_ids = []
+    for enrollment in user_enrollments:
+        enrollment_course_ids.append(enrollment.course_id)
+
+    for course in courses_in_program:
+        if not is_enrolled_in_course(course, enrollment_course_ids):
+            # User is not enrolled in this course, meaning they are not enrolled in all courses in the program
+            return False
+    # User is enrolled in all courses in the program
+    return True
+
+
+def is_enrolled_in_course(course, enrollment_course_ids):
+    """
+    Determine if the user is enrolled in this course
+    """
+    course_runs = course.get('course_runs')
+    if course_runs is None:
+        return False
+    for course_run in course_runs:
+        if is_enrolled_in_course_run(course_run, enrollment_course_ids):
+            return True
+    return False
+
+
+def is_enrolled_in_course_run(course_run, enrollment_course_ids):
```

Comment:
it looks like this is only used in the is_enrolled_in_course function. I would either include this code in the is_enrolled_in_course function, or define this function inside.